### ISSUE TITLE
Inconsistent weather data for Haarlem, the Netherlands

### ISSUE THREAD

**Issue by dzmitry-kankalovich** [chubin/wttr.in#854]
I've been using this service for my current city (Haarlem), but while I get the right values for the in-day breakdown, the summary looks totally out of place. 

I cross-referenced the data with other weather services and something doesn't add up. Check out the attached screenshot for example:
<img width="1274" alt="Screenshot 2023-02-27 at 10 16 08" src="https://user-images.githubusercontent.com/6346981/221529761-a90f057d-628a-4067-82e3-a33072efbfa8.png">

Maybe I don't get something, but the summary (or is it instant/real-time?) view is completely misaligned with the day forecast. It doesn't check with other weather services, and it's also plain wrong when I step outside the house literally.

I did check the weather in Harlem, New York and seemed close enough for me to suspect there might be some sort of a name collision?

**Comment by captn3m0** [comment]
Seeing the same issue with Bengaluru, India.

It showed up as Mist yesterday, and Haze today. Both around noon-time. Definitely incorrect.

![image](https://user-images.githubusercontent.com/584253/222064598-34ad2ee7-4ab5-41ae-b836-fc227003360c.png)

Ref: https://wttr.in/Bangalore

**Comment by dzmitry-kankalovich** [comment]
Seems to be the same case. Changed title to be more generic.

**Comment by dzmitry-kankalovich** [comment]
Potentially duplicate of https://github.com/chubin/wttr.in/issues/819

**Comment by OwlBawl** [comment]
same thing for Kyiv today.
Google, Apple, OpenWeather all just fine with 17-19 celsius
while gettin in console from request https://wttr.in/kyiv?format=%t;%h 
11*;66%

**Comment by sebastiancarlos** [comment]
Incorrect data for Los Angeles and other cities. I recommend everyone to stop using this application until the owner @chubin  addresses the issue.

<img width="916" alt="Screenshot 2023-05-27 at 07 21 52" src="https://github.com/chubin/wttr.in/assets/88276600/fffef119-b950-40ec-a187-a97f4bd26002">
<img width="805" alt="Screenshot 2023-05-27 at 07 21 31" src="https://github.com/chubin/wttr.in/assets/88276600/10029a17-540d-40f2-8f22-dab97e035b92">

**Comment by chubin** [comment]
People, don’t panic, that is just the data that we get from the upstream provider, and has with the app itself nothing to do. Use format=j1 option to see the raw data. Please let me know if the raw data is correct, and it is displayed incorrectly. We will try to fix the issue then. Otherwise, the only solution here is to drop this upstream provider and switch to another one. I already paid for the data for the many months upfront, so it won’t happen without serious arguments (and again: if the problem is not with the data, but with the app itself, please let me know)

**Comment by sebastiancarlos** [comment]
Thanks. I ceased to panic.

It is a problem with the upstream provider then, because the data is also off with `format=j1`

**Comment by kjetilho** [comment]
Don't know if it is helpful, but I see the same in Oslo:

```
Weather report: 59.92565,10.76668

     \  /       Partly cloudy
   _ /"".-.     +13(12) °C     
     \_(   ).   ← 19 km/h      
     /(___(__)  10 km          
                0.0 mm         
                                                       ┌─────────────┐                                                       
┌──────────────────────────────┬───────────────────────┤  Thu 15 Jun ├───────────────────────┬──────────────────────────────┐
│            Morning           │             Noon      └──────┬──────┘     Evening           │             Night            │
├──────────────────────────────┼──────────────────────────────┼──────────────────────────────┼──────────────────────────────┤
│    \  /       Partly cloudy  │    \  /       Partly cloudy  │  _`/"".-.     Thundery outbr…│    \  /       Partly cloudy  │
│  _ /"".-.     26 °C          │  _ /"".-.     +29(27) °C     │   ,\_(   ).   25 °C          │  _ /"".-.     16 °C          │
│    \_(   ).   ↓ 5-6 km/h     │    \_(   ).   ↙ 8-9 km/h     │    /(___(__)  ↗ 3-4 km/h     │    \_(   ).   → 1-3 km/h     │
│    /(___(__)  10 km          │    /(___(__)  10 km          │     ⚡‘‘⚡‘‘  9 km           │    /(___(__)  10 km          │
│               0.0 mm | 0%    │               0.0 mm | 0%    │     ‘ ‘ ‘ ‘   0.0 mm | 0%    │               0.0 mm | 0%    │
└──────────────────────────────┴──────────────────────────────┴──────────────────────────────┴──────────────────────────────┘
```

Current time is noon (12:16), so it is quite off as you can see (13 vs 29 degrees)

**Comment by uocnb** [comment]
2025-01-15T21:06:59+0700

Same in Hanoi, I included the moon day (lunar calendar) and it is off by one day (and possibly a half) - it is 16 in lunar calendar today as this comment written. The night temperature is around ~17-20°C

[https://vi.wttr.in/Hanoi?m&format="%l:+%c+%C+%t+%h+%w+%m+%M+%u"](https://vi.wttr.in/Hanoi?m&format=%l:+%c+%C+%t+%h+%w+%m+%M+%u)
```text
Hanoi: ☀️   Trời nắng +23°C 36% ↙10km/h 🌕 15 0
```

**Comment by chubin** [comment]
Yes, that is how the data is provided by the data source. What is possible though, as a workaround: we can add an option to take the current data from the forecast, but probably it will make the output less precise and not more, because the forecast is probably more off, than the current value.

Anyway, the problem is known, and this is a problem with the data source. Let's hope to solve it one day (buy adding more data sources).
